returnsrcvr-checker/src/main/java/org/checkerframework/checker/returnsrcvr/ReturnsRcvrAnnotatedTypeFactory.java

```diff
@@ -54,63 +63,40 @@ public Void visitExecutable(AnnotatedTypeMirror.AnnotatedExecutableType t, Void
-        AnnotatedTypeMirror.AnnotatedDeclaredType receiverType = t.getReceiverType();
-        receiverType.replaceAnnotation(THIS_ANNOT);
+      
+      // skip constructors 
+      if(!isConstructor(t)) {
+    	  // check each supported framework
+    	  for (FrameworkSupport frameworkSupport : frameworkSupports) {
+    		  // see if the method in the framework should return this
+            if (frameworkSupport.knownToReturnThis(t, (ReturnsRcvrAnnotatedTypeFactory)this.typeFactory)) {
+            	// add @This annotation
+            	  returnType.replaceAnnotation(THIS_ANNOT);
+                  AnnotatedTypeMirror.AnnotatedDeclaredType receiverType = t.getReceiverType();
+                  receiverType.replaceAnnotation(THIS_ANNOT);
+                  break;
+            }
+          }
       }
-
+         
       return super.visitExecutable(t, p);
     }
 
-    /**
-     * Checks if the given element is a setter in either an AutoValue builder or a Lombok builder
-     */
-    private boolean isBuilderSetter(AnnotatedTypeMirror.AnnotatedExecutableType t) {
-      ExecutableElement element = t.getElement();
-
-      // skip constructors
-      if (element.getKind().equals(ElementKind.CONSTRUCTOR)) {
-        return false;
-      }
-
-      Element enclosingElement = element.getEnclosingElement();
-
-      boolean inAutoValueBuilder = hasAnnotation(enclosingElement, AutoValue.Builder.class);
-      boolean inLombokBuilder =
-          (hasAnnotationByName(enclosingElement, "lombok.Generated")
-                  || hasAnnotationByName(element, "lombok.Generated"))
-              && enclosingElement.getSimpleName().toString().endsWith("Builder");
-
-      if (!inAutoValueBuilder && !inLombokBuilder) {
-        // see if superclass is an AutoValue Builder, to handle generated code
-        TypeMirror superclass = ((TypeElement) enclosingElement).getSuperclass();
-        // if enclosingType is an interface, the superclass has TypeKind NONE
-        if (!superclass.getKind().equals(TypeKind.NONE)) {
-          // update enclosingElement to be for the superclass for this case
-          enclosingElement = TypesUtils.getTypeElement(superclass);
-          inAutoValueBuilder = enclosingElement.getAnnotation(AutoValue.Builder.class) != null;
-        }
-      }
-
-      if (inAutoValueBuilder || inLombokBuilder) {
-        AnnotatedTypeMirror returnType = t.getReturnType();
-        return returnType != null
-            && enclosingElement.equals(TypesUtils.getTypeElement(returnType.getUnderlyingType()));
-      }
-
-      return false;
-    }
+  }
+  
+  private boolean isConstructor(AnnotatedTypeMirror.AnnotatedExecutableType t) {
+	  ExecutableElement element = t.getElement();
+	  return element.getKind().equals(ElementKind.CONSTRUCTOR);
   }
 
-  private boolean hasAnnotation(Element element, Class<? extends Annotation> annotClass) {
+  public boolean hasAnnotation(Element element, Class<? extends Annotation> annotClass) {
```

Comment:
These two methods should be `static`.  Or, @kelloggm, do you know if there are built-in methods in Checker Framework we can use instead of these?

Reply:
I don't know of such a convenience method in the framework itself. I agree these should be static.

Reply:
I tried to make them static but it seems they are using the non-static variables `elements` from `AnnotatedTypeFactory`, so I can't make them static.

Reply:
See below for suggestions on how to make the methods static